<launch>

<!-- Default folder with maps-->

<!-- <arg name="folder" default="$(env HOME)/eliska/maps/train"/> -->
<!-- <arg name="stc_file" default="$(env HOME)/history.txt"/> -->
<!-- <arg name="stc_model_type" default="Fremen"/> -->
<!-- <arg name="stc_model_param" default="43200"/> -->
<!-- <arg name="stc_strategy_type" default="Best"/> -->
<!-- <arg name="stc_strategy_param" default="500"/> -->


<!-- Distance Measurement Node -->
<node pkg="stroll_bearnav" type="odometry_monitor_node" name="odometry_monitor"/>

<!-- feature extraction -->
<node pkg="stroll_bearnav" type="feature_extraction_node" name="feature_extraction">
<remap from="image" to="/stereo/left/image_raw"/>
</node>

<node pkg="stroll_bearnav" type="mapper_node" name="mapper" output="screen">
<remap from="image" to="/stereo/left/image_raw"/>
<param name="folder" value="$(arg folder)" />
<param name="angularSpeed" value="1.1" />
</node>

<!-- map loader node -->
<node pkg="stroll_bearnav" type="map_preprocessor_node" name="map_preprocessor">
<param name="folder" value="$(arg folder)" />
<param name="stc_file" value="$(arg stc_file)" />
<param name="stc_model_type" value="$(arg stc_model_type)" />
<param name="stc_model_param" value="$(arg stc_model_param)" />
<param name="stc_strategy_type" value="$(arg stc_strategy_type)" />
<param name="stc_strategy_param" value="$(arg stc_strategy_param)" />
<remap from="map_preprocessor" to="map_preprocessor_map"/>

</node>

<!-- navigator node -->
<node pkg="stroll_bearnav" type="navigator_node" name="navigator" output="screen">
<param name="folder" value="$(arg folder)" />
</node>

</launch>
